### ISSUE TITLE
Issues with year 0 when plotting cftime with gregorian calendar

### ISSUE THREAD

**Issue by jbusecke** [SciTools/nc-time-axis#44]
I just ran into the problem I encountered over at [xarray/#3841](https://github.com/pydata/xarray/issues/3841). 

The basic problem is that if one plots a timeseries with cftime dates close to year 0 with the gregorian calendar, the [`NetCDFTimeDateLocator`](https://github.com/SciTools/nc-time-axis/blob/531dd0da83f2ea6b351f129559ccd4e8b0ccbb5f/nc_time_axis/__init__.py#L152-L165) seems to create a date that is crossing year 0, which is not defined.

I noticed I never actually opened an issue here, so here it is with a bit simpler example to reproduce the error:

```python
import numpy as np
import matplotlib.pyplot as plt
import nc_time_axis
import cftime

d_time = [cftime.DatetimeGregorian(year=year, month=1, day=1) for year in range(100, 131)]
temperatures = np.random.rand(len(d_time))

plt.plot(d_time, temperatures)
plt.show()
# this works
```
when I move the years closer to 0:
```python
d_time = [cftime.DatetimeGregorian(year=year, month=1, day=1) for year in range(1, 31)]
temperatures = np.random.rand(len(d_time))

plt.plot(d_time, temperatures)
plt.show()
# this gives the zero error.
```
I get the following error:
<details>

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-af84495616d7> in <module>
      3 
      4 plt.plot(d_time, temperatures)
----> 5 plt.show()
      6 # this gives the zero error.

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/matplotlib/pyplot.py in show(*args, **kw)
    270     """
    271     global _show
--> 272     return _show(*args, **kw)
    273 
    274 

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/ipykernel/pylab/backend_inline.py in show(close, block)
     41             display(
     42                 figure_manager.canvas.figure,
---> 43                 metadata=_fetch_figure_metadata(figure_manager.canvas.figure)
     44             )
     45     finally:

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/ipykernel/pylab/backend_inline.py in _fetch_figure_metadata(fig)
    178     if _is_transparent(fig.get_facecolor()):
    179         # the background is transparent
--> 180         ticksLight = _is_light([label.get_color()
    181                                 for axes in fig.axes
    182                                 for axis in (axes.xaxis, axes.yaxis)

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/ipykernel/pylab/backend_inline.py in <listcomp>(.0)
    181                                 for axes in fig.axes
    182                                 for axis in (axes.xaxis, axes.yaxis)
--> 183                                 for label in axis.get_ticklabels()])
    184         if ticksLight.size and (ticksLight == ticksLight[0]).all():
    185             # there are one or more tick labels, all with the same lightness

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/matplotlib/axis.py in get_ticklabels(self, minor, which)
   1318         if minor:
   1319             return self.get_minorticklabels()
-> 1320         return self.get_majorticklabels()
   1321 
   1322     def get_majorticklines(self):

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/matplotlib/axis.py in get_majorticklabels(self)
   1274     def get_majorticklabels(self):
   1275         'Return a list of Text instances for the major ticklabels.'
-> 1276         ticks = self.get_major_ticks()
   1277         labels1 = [tick.label1 for tick in ticks if tick.label1.get_visible()]
   1278         labels2 = [tick.label2 for tick in ticks if tick.label2.get_visible()]

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/matplotlib/axis.py in get_major_ticks(self, numticks)
   1429         'Get the tick instances; grow as necessary.'
   1430         if numticks is None:
-> 1431             numticks = len(self.get_majorticklocs())
   1432 
   1433         while len(self.majorTicks) < numticks:

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/matplotlib/axis.py in get_majorticklocs(self)
   1346     def get_majorticklocs(self):
   1347         """Get the array of major tick locations in data coordinates."""
-> 1348         return self.major.locator()
   1349 
   1350     def get_minorticklocs(self):

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/nc_time_axis/__init__.py in __call__(self)
    136     def __call__(self):
    137         vmin, vmax = self.axis.get_view_interval()
--> 138         return self.tick_values(vmin, vmax)
    139 
    140     def tick_values(self, vmin, vmax):

/scratch/gpfs2/jbusecke/conda_tigressdata/envs/euc_dynamics/lib/python3.8/site-packages/nc_time_axis/__init__.py in tick_values(self, vmin, vmax)
    192             raise ValueError(msg)
    193 
--> 194         return utime.date2num(ticks)
    195 
    196 

cftime/_cftime.pyx in cftime._cftime.utime.date2num()

cftime/_cftime.pyx in cftime._cftime.JulianDayFromDate()

cftime/_cftime.pyx in cftime._cftime._IntJulianDayFromDate()

ValueError: year zero does not exist in the standard calendar

</details>

Id be happy to submit a PR, but I would need some pointers on how to approach this problem best. 

My thinking was to check the generated vmin and vmax values [here](https://github.com/SciTools/nc-time-axis/blob/531dd0da83f2ea6b351f129559ccd4e8b0ccbb5f/nc_time_axis/__init__.py#L141) and set anything that is < 1 equal to 1? 

But perhaps a solution should be more general, and also allow all negative values? Do we need to make this dependent on the calendar, e.g. do some calendars allow a zero crossing?

**Comment by jbusecke** [comment]
This does solve the problem for me:

```python
def tick_values(self, vmin, vmax):
       ...
        ticks = [t for t in ticks if t.year != 0]
        return utime.date2num(ticks)
```

But that just leads to the ticks, not being displayed:
![image](https://user-images.githubusercontent.com/14314623/88347015-bd2ce780-cd17-11ea-9b72-867bfb04cf2a.png)

...which is not the prettiest...

**Comment by spencerkclark** [comment]
I would be +1 on a simple solution like the one you suggested @jbusecke; I can't really think of a clean way to automatically set the ticks when year zero is not allowed, so simply dropping it feels like a reasonable approach.  That way at least there wouldn't be an error; it would be nice if #41 were fixed in tandem to give the user finer-grained control of what the ticks were if they didn't like the default choice.

> Do we need to make this dependent on the calendar, e.g. do some calendars allow a zero crossing?

Indeed a number of calendars allow year zero -- all except `"gregorian"`, `"proleptic_gregorian"`, and `"julian"` -- so it would be good to continue to allow it for those calendars.

**Comment by jbusecke** [comment]
Should I put in a PR for this problem? Or would you prefer to get #41 done at the same time? I am not sure I have the bandwidth for both rn.

**Comment by spencerkclark** [comment]
Yeah, for sure, I would encourage you to.  If you want I can work on #41 separately, though I may not get to it until this weekend.

**Comment by jbusecke** [comment]
Also quite strapped for time. But i'll see what I can do!

**Comment by jbusecke** [comment]
Closed via #50
